### ISSUE TITLE
Issue on .ipynb demo: mismatched data shape?

### ISSUE THREAD

**Issue by MEtr0rite** [guanfangdong/LTS-Learning-Temporal-Distribution-and-Spatial-Correlation-Towards-Universal-MOS#2]
I'm not sure about the reason of this issue, but there is a mismatch of data shape in the demo of DIDL model, where the groundtruth mask is read by method `readImg_byFilesIdx_pytorch()`. The result `labs_tru` is read as an RGB image, with shape (H,W,3). Since the result of `detectFg()` Is a mask with shape (H,W), There should be an adjustment made here to align them with each other. The current code will lead to an RE in evaluation stage (method `evaluation_numpy_entry_torch()` ).

![Image](https://github.com/user-attachments/assets/306e9aec-14b6-4080-9656-5d2037a9ee1c)

The involve code is attached below. I've made a change(`gt_fg = np.round(labs_tru)[:, :, 0]`) to solve the issue:

    if not os.path.exists(check_filename):
        data = eximgs[frame_idx] / 255.0
        subhist_sp = getSubHists_byHist(data, hists_tensor_sp)
        size = subhist_sp.shape
        subhist_sp = subhist_sp.reshape(size[0], size[1]*size[2], size[3]*size[4], size[5]).permute(1, 0, 2, 3)
        re_labs = detectFg(subhist_sp, batch_size, device, X, W, B, F_W, F_B, proddis, sumdis, netP, netC) / 2
        im_fg = np.round(re_labs.reshape(row_im, column_im) * 255)
    else:
        im_fg = imageio.imread(check_filename)
    labs_tru = readImg_byFilesIdx_pytorch(frame_idx, path_config['gt'][1], path_config['gt'][0]).numpy()
    gt_fg = np.round(labs_tru)[:,:,0]
    TP, FP, TN, FN = evaluation_numpy_entry_torch(torch.tensor(im_fg), torch.tensor(gt_fg))

If you have the time, please check out this part of your code to make sure everything goes soomthly. Maybe a mechanism of checking dimension of data like what you've done in histogram generation could help:
        
    imgs = loadImgs_pytorch(path_config['im'][1], path_config['im'][0])
    print(f"Original images shape: {imgs.shape}")
    if imgs.ndim == 3: 
        imgs = imgs.unsqueeze(3).repeat(1, 1, 1, 3)
